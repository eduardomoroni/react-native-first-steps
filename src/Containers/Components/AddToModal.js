import React, { Component } from 'react'
import { View, StyleSheet } from 'react-native'
import PropTypes from 'prop-types'
import { cardType } from '../../Types/CardType'
import { Modal } from './'

import { RkText } from 'react-native-ui-kitten'
import { Button } from 'react-native-elements'

export class AddToModal extends Component {
  constructor() {
    super()
    this.state = {
      isVisible: true,
    }
  }

  toggleModal () {
    this.state.isVisible = !this.state.isVisible
  }

  render () {
    const { card, list } = this.props
    return (
      <Modal isVisible onModalHide={() => this.toggleModal()} >
        <View style={styles.modal}>
          <RkText style={styles.title}>
            {`ADD ${card.name} TO WANT LIST`}
          </RkText>
          <Button
            large
            iconRight
            icon={{name: 'code'}}
            onPress={() => console.log('ADD CARD TO SOMETHING')}
            title='ADD'
          />
        </View>
      </Modal>
    )
  }
}

AddToModal.propTypes = {
  card: cardType.isRequired,
  listToAdd: PropTypes.any
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  title: {
    fontSize: 18,
    textAlign: 'center'
  },
  modal: {
    backgroundColor: 'white',
    height: 400
  }
})
