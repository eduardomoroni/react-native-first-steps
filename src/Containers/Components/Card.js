/* @flow */

import React, { Component } from 'react'
import {
  View,
  Text,
  StyleSheet
} from 'react-native'

type CardProps = {
  card: any
}

export default class Card extends Component {
  props: CardProps
  render () {
    const {
      name,
      text
    } = this.props.card
    return (
      <View style={styles.container}>
        <Text>{name}</Text>
        <Text>{text}</Text>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  }
})
