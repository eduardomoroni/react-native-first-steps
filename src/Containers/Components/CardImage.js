/* @flow */

import React, { PureComponent } from 'react'
import {
  View,
  StyleSheet,
  Image,
  ActivityIndicator
} from 'react-native'

type CardImageProps = {
  card: any
}

type CardImageState = {
  loading: boolean
}

export class CardImage extends PureComponent {
  props: CardImageProps
  state: CardImageState

  constructor () {
    super()
    this.state = { loading: true }
  }

  render () {
    const { multiverseid } = this.props.card
    const IMG_URL = `http://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=${multiverseid}&type=card`

    return (
      <View style={styles.container}>
        <ActivityIndicator size='large' style={this.state.loading ? styles.spinner : styles.hidden} />
        <Image
          style={this.state.loading ? styles.hidden : styles.image}
          source={{uri: IMG_URL}}
          onLoad={(e) => this.setState({loading: false})}
          onLoadStart={(e) => this.setState({loading: true})}
          onError={() => console.error('Error while loading card image')} />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  image: {
    flex: 1
  },
  spinner: {
    flex: 1,
    justifyContent: 'center'
  },
  hidden: {
    width: 0,
    height: 0
  }
})
