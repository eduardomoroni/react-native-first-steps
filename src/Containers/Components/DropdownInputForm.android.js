/* @flow */

import React, { Component } from 'react'
import { View } from 'react-native'
import { InputLabel, InputPicker } from './'
import styles from '../../Styles/FormStyle'

export default class DropdownInputForm extends Component {
  render () {
    const { input, dropdownItems, selectedValue } = this.props
    const { onChange, name } = input

    return (
      <View style={styles.container}>
        <InputLabel label={name} />
        <InputPicker
          selectedValue={selectedValue}
          onValueChange={onChange}
          dropdownItems={dropdownItems}
        />
      </View>
    )
  }
}
