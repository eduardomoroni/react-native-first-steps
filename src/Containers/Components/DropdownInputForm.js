/* @flow */

import React, { Component } from 'react'
import I18n from 'react-native-i18n'
import {
  View,
  Text,
  Picker
} from 'react-native'
import styles from '../../Styles/FormStyle'

export default class DropdownInputForm extends Component {
  render () {
    const { input, dropdownItems, selectedValue } = this.props
    const { onChange, name } = input

    const renderPickerItem = (value, key) => {
      return <Picker.Item label={value} value={value} key={key} />
    }

    return (
      <View style={styles.container}>
        <Text style={styles.text}>
          {I18n.t(name)}
        </Text>
        <Picker
          style={styles.dropdown}
          selectedValue={selectedValue}
          onValueChange={onChange}>
          { dropdownItems.map(renderPickerItem) }
        </Picker>
      </View>
    )
  }
}
