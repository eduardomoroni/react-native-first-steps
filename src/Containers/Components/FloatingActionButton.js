import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { StyleSheet } from 'react-native'
import ActionButton from 'react-native-action-button'
import Icon from 'react-native-vector-icons/Ionicons'

export class FloatingActionButton extends Component {
  renderItem (item: ItemType, key: number) {
    const { name, ...itemProps } = item

    return (
      <ActionButton.Item {...itemProps} key={key}>
        <Icon name={name} style={styles.actionButtonIcon} />
      </ActionButton.Item>
    )
  }

  render () {
    return (
      <ActionButton buttonColor='rgba(231,76,60,1)'>
        {this.props.items.map(this.renderItem)}
      </ActionButton>
    )
  }
}

const ItemType = {
  name: PropTypes.string.isRequired,
  buttonColor: PropTypes.string,
  title: PropTypes.string,
  onPress: PropTypes.func
}

FloatingActionButton.propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape(ItemType))
}

const styles = StyleSheet.create({
  actionButtonIcon: {
    fontSize: 20,
    height: 22,
    color: 'white'
  }
})
