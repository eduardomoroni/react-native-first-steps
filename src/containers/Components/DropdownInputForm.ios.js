/* @flow */

import React, { Component } from 'react'
import { View, TextInput } from 'react-native'
import { InputLabel } from './'
import styles from '../../styles/FormStyle'

// WIP
export class DropdownInputForm extends Component {
  render () {
    const { onChange, name } = this.props.input

    return (
      <View style={styles.container}>
        <InputLabel label={name} />
        <TextInput
          style={styles.input}
          onChangeText={onChange}
          autoCorrect={false}
          autoCapitalize='characters'
        />
      </View>
    )
  }
}

export default DropdownInputForm
