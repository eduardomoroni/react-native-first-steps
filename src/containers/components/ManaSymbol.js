/* @flow */

import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import Mana from '../../assets/mana'
import { Image } from 'react-native'

export const ValidColors = ['black', 'blue', 'green', 'red', 'white']

export class ManaSymbol extends PureComponent {
  render () {
    return <Image style={this.props.style} source={Mana[this.props.color]} />
  }
}

ManaSymbol.propTypes = {
  style: PropTypes.any,
  color: PropTypes.oneOf(ValidColors).isRequired
}
