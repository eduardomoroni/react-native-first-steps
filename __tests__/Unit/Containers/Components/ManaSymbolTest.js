// @flow

import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import { ManaSymbol, ValidColors } from '../../../../src/Containers/Components/ManaSymbol'

const props = {
  color: ValidColors[2],
  style: {backgroundColor: 'blue'}
}

it('Should render TextInputForm Component', () => {
  const tree = shallow(<ManaSymbol {...props} />)
  expect(tree).toMatchSnapshot()
})
