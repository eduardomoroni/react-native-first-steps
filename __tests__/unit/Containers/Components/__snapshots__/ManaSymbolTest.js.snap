// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ManaSymbol /> Snapshot 1`] = `
<Image
  source={1}
  style={
    Object {
      "backgroundColor": "blue",
    }
  }
/>
`;
